modules/tensor_mechanics/src/materials/RadialReturnStressUpdate.C

```diff
@@ -90,13 +97,24 @@ RadialReturnStressUpdateTempl<is_ad>::calculateNumberSubsteps(
 
   if (!MooseUtils::absoluteFuzzyEqual(effective_elastic_strain, 0.0))
   {
-    const Real ratio = effective_elastic_strain / _max_inelastic_increment;
 
-    if (ratio > _substep_tolerance)
-      substep_number = std::ceil(ratio / _substep_tolerance);
+    if (!_use_substep_integration_error)
+    {
+      const Real ratio = effective_elastic_strain / _max_inelastic_increment;
+
+      if (ratio > _substep_tolerance)
+        number_of_substeps = std::ceil(ratio / _substep_tolerance);
+    }
+    else
+    {
+      const Real accurate_time_step = _dt * _substep_tolerance / effective_elastic_strain;
+
+      if (_dt > accurate_time_step)
+        number_of_substeps = std::ceil(_dt / accurate_time_step);
```

Comment:
It seems like this can be refactored so that we do not need `_dt` here:

```
      const Real ratio = _substep_tolerance / effective_elastic_strain;

      if (_ratio<1.0)
        number_of_substeps = std::ceil(1.0 / ratio);
```